### PR TITLE
fix: double msal issue

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

- [x] Removed old MSAL solution
- [x] Rewrite everything to fusion framework
- [ ] Hopefully fixed everything but needs thorough testing

### CODE REVIEW

**Review by jaysencpp — COMMENTED** [review]
some comments

File: src/components/Routes/Routes.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ export function ClientRoutes(): JSX.Element {
     const {
         appConfig,
         registry: { apps, appGroups },
-        internal: { authProvider },
+        // internal: { authProvider },
```
Comment (hakonrossebo) [inline src/components/Routes/Routes.tsx:56]: Fjerne?

**Review by hakonrossebo — APPROVED** [review]

File: src/AppClient.tsx [hunk 2]
```diff
@@ -1,63 +1,171 @@
...
+            staleTime: Infinity,
+            cacheTime: Infinity,
+            suspense: true,
+            useErrorBoundary: true,
+        }
+    );
+
+    if (!data) {
+        throw new Error('Portal failed to setup');
+    }
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/AppClient.tsx:75]: suspense and boundary

File: src/AppClient.tsx [hunk 3]
```diff
@@ -1,63 +1,171 @@
...
+    const contextClient = useHttpClient('fusionContext');
+
+    const { isLoading } = useQuery(['setup'], async () => {
+        registerClientRegistry(
+            setupApps(appsProvider(getApps, getAppGroups, false), appConfig, client)
+        );
+        await setupContext(contextClient);
+
+        //TODO: how to handle this?
+        // setupUserData(authProvider);
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/AppClient.tsx]: From framework and people api

File: src/AppClient.tsx [hunk 4]
```diff
@@ -1,63 +1,171 @@
...
+                    </BrowserRouter>
+                </MenuProvider>
+            </ClientContextProvider>
+        </>
     );
 };
 
-export default Client;
+const useSetupClients = () => {
+    const isMounted = useRef(false);
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/AppClient.tsx:127]: The hook is being used?

**Review by jaysencpp — APPROVED** [review]
Need to test everything in dev ;)

File: src/apps/Handover/Garden/hooks/useHandoverResource.ts [hunk 5]
```diff
@@ -21,18 +21,18 @@ const useHandoverResource = <T extends keyof HandoverResourceTypeMap>(
     This causes an infinite render loop when added as dependency to getData.
      apiClient should be stable, but does not look to be the case.   
     **/
-    const { fusionDataproxy } = useHttpClient();
+    const client = useHttpClient('fusionDataproxy');
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/apps/Handover/Garden/hooks/useHandoverResource.ts:24]: Yep

File: src/AppClient.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jaysencpp) [reply src/AppClient.tsx]: const isMounted

File: src/components/TopBar/TopBarAvatar.tsx [hunk 6]
```diff
@@ -17,21 +25,26 @@ export const TopBarAvatar = (): JSX.Element | null => {
...
-    const { getCurrentUser } = useAuthProvider();
+    const client = useHttpClient('fusionPeople');
 
     const { data: presence } = useQuery(
         PresenceQueryKey,
-        async () => await getUserPresence(getCurrentUser()?.localAccountId.split('.')[0] ?? ''),
+        async () => {
+            return (
+                await client.fetch(`persons/${user?.localAccountId.split('.')[0] ?? ''}/presence`)
+            ).json();
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/components/TopBar/TopBarAvatar.tsx:32]: Your personal azure oid combined with some other stuff

File: src/packages/BookmarksManager/src/utils/api/getAllBookmarks.ts [hunk 7]
```diff
@@ -6,7 +6,7 @@ type ErrorResponse = {
 export const getAllBookmarks = async (signal?: AbortSignal) => {
     const { fusionBookmarks } = httpClient();
     const filterSourceSystem = "$filter=sourcesystem.name eq 'Castberg-portal'";
```
[the remark this answers is not in the document]
Comment (Gustav-Eikaas) [reply src/packages/BookmarksManager/src/utils/api/getAllBookmarks.ts:8]: The preferred way is useHttpClient but its so many files affected, I might do it in another pr

**Pull request merged**